lib/tile_bag.rb

```diff
@@ -0,0 +1,63 @@
+class TileBag
+  attr_accessor :tile_hash, :tile_bag
+  def initialize
+    @tile_hash = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1
+    }
+    @tile_bag = []
+    #@tile_tray = []
+    self.make_tiles
+  end
+
+#makes an instance of a tile_bag that initially includes all the letters possible in the game
+  def make_tiles
```

Comment:
This is a good use of both the hash datatype to store the tile count of each letter initially, and the array datatype to keep track of the set of actual tiles stored in an individual bag. Both datatypes are being used in a way that makes the most of their unique benefits.